test/easyconfigs/easyconfigs.py

```diff
@@ -1429,7 +1430,13 @@ def template_easyconfig_test(self, spec):
                 # 4th value is toolchain spec
                 if len(dumped_dep) >= 4:
                     if len(orig_dep) >= 4:
-                        self.assertEqual(dumped_dep[3], orig_dep[3])
+                        # if True was used to indicate that dependency should use system toolchain,
+                        # then we need to compare the value for the dumped easyconfig more carefully;
+                        # see also https://github.com/easybuilders/easybuild-framework/pull/4069
+                        if orig_dep[3] == True:
```

Comment:
```suggestion
                        if orig_dep[3] is True:
```

Reply:
i guess this is the nicest in this setting, don't want implicit true conversions i think